### PR TITLE
[ml][ai] Turn on docstring validation

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

This PR turns on the strict_sphinx docstring validation for the azure-ai-generative and azure-ai-resources packages, including fixing all existing errors to get the documentation builds in compliance. It also makes significant progress in fixing the documentation build errors for azure-ai-ml, but not all errors can be addressed at this time, so the strict_sphinx flag will be set to true in a future PR.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py [hunk 1]
```diff
@@ -403,8 +403,8 @@ def begin_cancel(self, name: str, **kwargs: Any) -> LROPoller[None]:
         .. admonition:: Example:
 
             .. literalinclude:: ../samples/ml_samples_misc.py
-                :start-after: [START job_operations_begin_cancel]
-                :end-before: [END job_operations_begin_cancel]
+                :start-after: [START job_operations_cancel]
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:406]: Why is this being changed? The function name is `begin_cancel` so it's unclear to me why we're removing the `begin` from this.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 2]
```diff
@@ -87,7 +87,9 @@ class ModelOperations(_ScopeDependentOperations):
     :type service_client: typing.Union[
         ~azure.ai.ml._restclient.v2023_04_01_preview._azure_machine_learning_workspaces.AzureMachineLearningWorkspaces,
         ~azure.ai.ml._restclient.v2021_10_01_dataplanepreview._azure_machine_learning_workspaces.
-        AzureMachineLearningWorkspaces]
+        AzureMachineLearningWorkspaces
+
+    ]
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:92]: Is this whitespace needed? Seems superfluous.

File: sdk/ml/azure-ai-ml/pyproject.toml [hunk 3]
```diff
@@ -4,16 +4,8 @@ pyright = false
 type_check_samples = false
 verifytypes = false
 pylint = true
+strict_sphinx = false
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/pyproject.toml:7]: Sphinx false? I thought we wanted to keep sphinx active in our repos. Is this not related to that?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:406]: Good catch - that shouldn't have been changed!

Also troubling that Sphinx doesn't catch that...

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:92]: Yes, annoyingly Sphinx gets confused with listing things inside brackets/parentheses. The only way I could get it to stop complaining about no line after was to add this spacing. From the docs I generated locally, it shouldn't affect the appearance in our reference docs.

File: sdk/ml/azure-ai-ml/pyproject.toml (hunk 3 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/pyproject.toml:7]: I noted this in the PR description, but even though this PR fixes Sphinx errors in azure-ai-ml, azure-ai-generative, and azure-ai-resources, I was only able to turn strict_sphinx = true for the azure-ai-generative and azure-ai-resources packages because azure-ai-ml has some other errors that are currently unfixable.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:406]: Ah - actually upon looking further this was right. The person who added the example code in samples/ that's referenced here, named it `job_operations_cancel` instead of `job_operations_begin_cancel` but wrote `job_operations_begin_cancel`, so the sample was referring to an example that actually didn't exist. I just changed the name here to match the name in the sample. To make it clearer though, I'll just change the name of the sample to `job_operations_begin_cancel` and match it here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py]: Update: I found out another way to fix the error that allowed me to remove the whitespace :) 

I confirmed it looks correct in the doc build.

**Review by MilesHolland — APPROVED** [review]
